### PR TITLE
fix: callback body validation errors

### PR DESCRIPTION
**Pull request by artursudnik**
This PR:

- adds logic that throws an exception after callback body validation returns errors
- fixes bugs causing validation errors for a valid callback body
- fixes nested objects not validated in DTOs
- fixes tests failing after fixing DTOs validations
- fixes Presentation.verifiableCredential defined as mandatory, but it is actually optional according to the [vc-data-model](https://www.w3.org/TR/vc-data-model/#presentations-0)

### CODE REVIEW

File: apps/vc-api/src/vc-api/exchanges/exchange.service.ts [hunk 1]
```diff
@@ -122,8 +124,17 @@ export class ExchangeService {
...
+    });
+
+    if (validationErrors.length > 0) {
+      this.logger.error('\n' + validationErrors.map((e) => e.toString()).join('\n\n'));
+      throw new Error(validationErrors.toString());
+    }
+
     callback?.forEach((callback) => {
       this.httpService.post(callback.url, body).subscribe({
         next: (v) => Logger.log(inspect(v)), // inspect used to replace circular references https://stackoverflow.com/a/18354289
```
Comment (jrhender) [inline apps/vc-api/src/vc-api/exchanges/exchange.service.ts]: I know it's not clearly part of this change, but I think we should use the new logger instantiation here as well, right?
```suggestion
        next: (v) => this.logger.log(inspect(v)), // inspect used to replace circular references https://stackoverflow.com/a/18354289
```

File: tests/e2e/src/main-suite.e2e-spec.ts [hunk 2]
```diff
@@ -129,6 +129,12 @@ describe('E2E Suite', function () {
     const validInputDescriptor = {
       constraints: {
         fields: [
+          {
+            path: ['$.id'],
+            filter: {
+              const: 'https://issuer.oidp.uscis.gov/credentials/83627465'
```
Comment (jrhender) [inline tests/e2e/src/main-suite.e2e-spec.ts]: Perhaps a minor point but this `.gov` URL as the id is a bit out of context here so better just to use a UUID, in my opinion
```suggestion
              const: 'urn:uuid:49f69fb8-f256-4b2e-b15d-c7ebec3a507e'
```

**Review by jrhender — APPROVED** [review]
Some minor suggestions but it looks good to me, thanks

File: apps/vc-api/src/vc-api/exchanges/exchange.service.ts (hunk 1 above)
Comment (artursudnik) [reply apps/vc-api/src/vc-api/exchanges/exchange.service.ts]: 100% agree! thanks!

File: apps/vc-api/src/vc-api/credentials/dtos/presentation.dto.ts [hunk 3]
```diff
@@ -57,5 +58,6 @@ export class PresentationDto implements Presentation {
    * The Verifiable Credentials
    */
   @ValidateNested()
+  @Type(() => VerifiableCredentialDto)
```
Comment (JGiter) [inline apps/vc-api/src/vc-api/credentials/dtos/presentation.dto.ts:61]: Should we also add @IsArray? Also is @ValidateNested will check verifiableCredential or each element in verifiableCredential? If I remember correctly I added `each` = true to @ValidateNested config in such cases

Comment (artursudnik) [reply apps/vc-api/src/vc-api/credentials/dtos/presentation.dto.ts:61]: I 100% agree. I made changes in bulk when fixing it. Thanks!

However, now I see that tests need to be adjusted. They fail after this change.

**Pull request merged**